lib/node_modules/@stdlib/blas/base/scnrm2/lib/ndarray.js

```diff
@@ -0,0 +1,87 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var absf = require( '@stdlib/math/base/special/absf' );
+var imagf = require( '@stdlib/complex/imagf' );
+var realf = require( '@stdlib/complex/realf' );
+var sqrtf = require( '@stdlib/math/base/special/sqrtf' );
+
+
+// MAIN //
+
+/**
+* Finds the euclidean norm of a complex single-precision floating-point vector.
+*
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Complex64Array} cx - input array
+* @param {NonNegativeInteger} strideX - `cx` stride length
+* @param {NonNegativeInteger} offsetX - starting index for `cx`
+* @returns {number} index value
+*
+* @example
+* var Complex64Array = require( '@stdlib/array/complex64' );
+*
+* var cx = new Complex64Array( [ 0.3, 0.1, 0.5, 0.0, 0.0, 0.5, 0.0, 0.2, 2.0, 3.0 ] );
+*
+* var norm = scnrm2( 4, cx, 1, 0 );
+* // returns ~0.8
+*/
+function scnrm2( N, cx, strideX, offsetX ) {
+	var scale;
+	var norm;
+	var temp;
+	var ssq;
+	var ix;
+
+	if ( N < 1 || strideX < 0 ) {
```

Comment:
`strideX < 0` does not follow more modern BLAS implementations. See https://www.netlib.org/lapack/explore-html/d1/d2a/group__nrm2_gaee5779d5d216a7cd8cf83488fb6bb175.html#gaee5779d5d216a7cd8cf83488fb6bb175.

Reply:
We should update our real implementations, as well.